systemd/escape.go

```diff
@@ -61,9 +71,21 @@ func EscapeUnitNamePath(in string) string {
 		} else if strings.IndexByte(allowed, c) >= 0 {
 			buf.WriteByte(c)
 		} else {
-			fmt.Fprintf(buf, `\x%x`, []byte{in[i]})
+			fmt.Fprintf(&buf, `\x%x`, []byte{in[i]})
 		}
 	}
 
 	return buf.String()
 }
+
+// EscapeUnitNamePath works like systemd-escape --path
+// FIXME: we could use github.com/coreos/go-systemd/unit/escape.go and EscapePath
+// from it. But that's not in the archive and it won't work with go1.3
```

Comment:
hm maybe we should just switch to using go-systemd? It's already an indirect dependency of ours. From the looks of it debian has v22 https://packages.debian.org/search?suite=all&searchon=names&keywords=go-systemd and so does Fedora https://koji.fedoraproject.org/koji/packageinfo?packageID=29480 and there's probably other places we could use it too